compiler/testData/ir/irText/types/castsInsideCoroutineInference.fir.txt

```diff
@@ -39,7 +39,7 @@ FILE fqName:<root> fileName:/castsInsideCoroutineInference.kt
               $receiver: VALUE_PARAMETER name:<this> type:<root>.FlowCollector<T of <root>.onCompletion>
               BLOCK_BODY
                 VAR name:safeCollector type:<root>.SafeCollector<IrErrorType> [val]
-                  CONSTRUCTOR_CALL 'public constructor <init> (collector: <root>.FlowCollector<T of <root>.SafeCollector>) [primary] declared in <root>.SafeCollector' type=<root>.SafeCollector<IrErrorType> origin=null
+                  CONSTRUCTOR_CALL 'public constructor <init> (collector: <root>.FlowCollector<T of <root>.SafeCollector>) [primary] declared in <root>.SafeCollector' type=<root>.SafeCollector<T of <root>.onCompletion> origin=null
```

Comment:
Ah, there are more `IrErrorType` in this file already. 🤦

Reply:
There are still some remaining `IrErrorType` in this case. Maybe the scope for stub type substitutor should be broader than what it covers for now: types in completed calls inside builder-inferred lambda. For example, `val safeCollector = ...` inside the lambda has a substituted type for the assigning expression, but its variable type is not substituted (since it's not a call). 

Another one above as type operand seems more complicated. It came from smartcast info, so any evidences inside type constraint system need some updates as well. Both seems out of scope of this PR, though.

Reply:
I think it's fine to call `FirStubTypeTransformer` over whole top-level call

Reply:
I left a TODO at `updateCalls` to explicitly mention that we need to update more than just calls.